homeassistant/components/ipp/config_flow.py

```diff
@@ -0,0 +1,160 @@
+        except IPPConnectionUpgradeRequired:
+            return self._show_setup_form({"base": "connection_upgrade"})
+        except IPPConnectionError:
+            return self._show_setup_form({"base": "connection_error"})
+        user_input[CONF_UUID] = info[CONF_UUID]
+
+        await self.async_set_unique_id(user_input[CONF_UUID])
+        self._abort_if_unique_id_configured(updates={CONF_HOST: user_input[CONF_HOST]})
+
+        return self.async_create_entry(title=user_input[CONF_HOST], data=user_input)
+
+    async def async_step_zeroconf(self, discovery_info: ConfigType) -> Dict[str, Any]:
+        """Handle zeroconf discovery."""
+        # Hostname is format: EPSON123456.local.
+        host = discovery_info["hostname"].rstrip(".")
+        port = discovery_info["port"]
+        name, _ = host.rsplit(".")
+        tls = discovery_info["type"] == "_ipps._tcp.local."
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update({"title_placeholders": {"name": name}})
+
+        self.discovery_info.update(
+            {
+                CONF_HOST: host,
+                CONF_PORT: port,
+                CONF_SSL: tls,
+                CONF_VERIFY_SSL: False,
+                CONF_BASE_PATH: "/"
+                + discovery_info["properties"].get("rp", "ipp/print"),
+                CONF_NAME: name,
+                CONF_UUID: discovery_info["properties"].get("UUID"),
+            }
+        )
+
+        if self.discovery_info[CONF_UUID] is None:
+            try:
+                info = await validate_input(self.hass, self.discovery_info)
+            except IPPConnectionUpgradeRequired:
+                return self.async_abort(reason="connection_upgrade")
+            except IPPConnectionError:
+                return self.async_abort(reason="connection_error")
+            self.discovery_info[CONF_UUID] = info[CONF_UUID]
+
+        await self.async_set_unique_id(self.discovery_info[CONF_UUID])
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self.discovery_info[CONF_HOST]}
+        )
+
+        return await self.async_step_zeroconf_confirm()
+
+    async def async_step_zeroconf_confirm(
+        self, user_input: ConfigType = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by zeroconf."""
+        if user_input is None:
+            return self._show_confirm_dialog()
+
+        try:
+            info = await validate_input(self.hass, self.discovery_info)
+        except IPPConnectionUpgradeRequired:
+            return self.async_abort(reason="connection_upgrade")
+        except IPPConnectionError:
+            return self.async_abort(reason="connection_error")
+        self.discovery_info[CONF_UUID] = info[CONF_UUID]
+
+        await self.async_set_unique_id(self.discovery_info[CONF_UUID])
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self.discovery_info[CONF_HOST]}
+        )
```

Comment:
Already done in step_confirm.

Reply:
@balloob so I removed from here and left in confirm but should it go before the form showing to avoid asking to confirm something that already exists?